sdk/helper/storagepacker/storagepacker_v2.go

```diff
@@ -162,59 +268,51 @@ func (s *StoragePackerV2) BucketKeys(ctx context.Context) ([]string, error) {
 
 // Get returns a bucket for a given key
 func (s *StoragePackerV2) GetBucket(ctx context.Context, key string, skipCache bool) (*LockedBucket, error) {
-	cacheKey := s.GetCacheKey(key)
-
 	if key == "" {
-		return nil, fmt.Errorf("missing bucket key")
+		return nil, errors.New("missing bucket key")
 	}
 
-	lock := locksutil.LockForKey(s.storageLocks, cacheKey)
-	lock.RLock()
-
-	s.bucketsCacheLock.RLock()
-	_, bucketRaw, found := s.bucketsCache.LongestPrefix(cacheKey)
-	s.bucketsCacheLock.RUnlock()
-
-	if found && !skipCache {
-		ret := bucketRaw.(*LockedBucket)
-		lock.RUnlock()
-		return ret, nil
+	if !skipCache {
+		bucket, found, err := s.lockBucketForRead(key)
+		if err != nil {
+			return nil, err
+		}
+		if found {
+			bucket.RUnlock()
+			return bucket, nil
+		} else {
+			return nil, nil
+		}
 	}
 
-	// Swap out for a write lock
-	lock.RUnlock()
-	lock.Lock()
-	defer lock.Unlock()
-
-	// Check for it to have been added
-	s.bucketsCacheLock.RLock()
-	_, bucketRaw, found = s.bucketsCache.LongestPrefix(cacheKey)
-	s.bucketsCacheLock.RUnlock()
-
-	if found && !skipCache {
-		ret := bucketRaw.(*LockedBucket)
-		return ret, nil
+	// Read directly from storage
+	// FIXME: in the cases where we want skipCache = true, why do we trust
```

Comment:
As far as I can tell the only place skipCache is set true is for some invalidation code -- I think it can/will actually go away.